src/lib/format/l20n/ast/parser.js

```diff
@@ -101,16 +104,16 @@ class ParseContext {
     return entity;
   }
 
-  getValue(ch = this._source[this._index], optional = false) {
+  getValue(ch = this._source[this._index], index = false, required = true) {
```

Comment:
OK, I see, thanks for the explanation.  One additional nit:  perhaps rename the arguments to `hasIndex` and `isRequired`?